pallets/automation-time/src/lib.rs

```diff
@@ -606,12 +606,14 @@ pub mod pallet {
 			let who = ensure_signed(origin)?;
 			let provided_id: Vec<u8> =
 				Self::generate_auto_compound_delegated_stake_provided_id(&who, &collator_id);
-			if frequency % 3600 != 0 {
-				Err(Error::<T>::InvalidTime)?
-			}
-			if frequency > T::MaxScheduleSeconds::get() {
-				Err(Error::<T>::TimeTooFarOut)?
+
+			let next_execution =
+				execution_time.checked_add(frequency).ok_or(Error::<T>::TimeTooFarOut)?;
+			Self::is_valid_time(next_execution)?;
```

Comment:
If I'm reading correctly you did it this way to both validate the `execution_time` and `frequency` at the same time?  You are ensuring `frequency` is not `0` and also that the `execution_time` is valid (`% 3600`).  It might be helpful to add a comment what you are trying to accomplish.  Might not be clear to someone in the future.

Reply:
Execution time is validated later on. This is to validate frequency without having to rewrite the valid_time code. Added a comment to that effect.